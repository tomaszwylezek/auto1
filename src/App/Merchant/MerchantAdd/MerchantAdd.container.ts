import { connect } from 'react-redux';

import { MerchantAdd } from './MerchantAdd';

import { addMerchant } from 'redux/merchants/merchants.actions';

import { IMerchantAddConnectedActions } from './MerchantAdd.types';

const mapDispatchToProps: IMerchantAddConnectedActions = {
  addMerchant
};

export const MerchantAddContainer = connect(
  null,
  mapDispatchToProps
)(MerchantAdd);
