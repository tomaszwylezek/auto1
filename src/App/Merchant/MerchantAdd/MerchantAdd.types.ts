import { IMerchant } from 'model';

export interface IMerchantAddOwnProps {}

export interface IMerchantAddConnectedState {}

export interface IMerchantAddConnectedActions {
  addMerchant: (merchant: IMerchant) => void
}

export interface IMerchantAddProps extends IMerchantAddConnectedState, IMerchantAddConnectedActions, IMerchantAddOwnProps {}

export interface IMerchantAddState {}

