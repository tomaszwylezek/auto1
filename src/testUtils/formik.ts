import { FormikActions, FormikHandlers, FormikProps } from 'formik';
import { FormikField } from 'model';

export function getMockedFormikConfig<T>(options: Partial<FormikProps<T>> = {}): FormikProps<T> {
  return {
    ...getMockedFormikActions(),
    ...getMockedFormikHandles(),
    enableReinitialize: false,
    isValidating: false,
    isSubmitting: false,
    dirty: false,
    isValid: false,
    submitCount: 0,
    registerField: jest.fn(),
    unregisterField: jest.fn(),
    values: {} as T,
    errors: {} as T,
    initialValues: {} as T,
    touched: {} as T,
    ...options
  };
}

export function getMockedFormikActions<V>(options: Partial<FormikActions<V>> = {}): FormikActions<V> {
  return {
    setStatus: jest.fn(),
    setError: jest.fn(),
    setErrors: jest.fn(),
    setSubmitting: jest.fn(),
    setTouched: jest.fn(),
    setValues: jest.fn(),
    setFieldValue: jest.fn(),
    setFieldError: jest.fn(),
    setFieldTouched: jest.fn(),
    validateForm: jest.fn(),
    validateField: jest.fn(),
    resetForm: jest.fn(),
    submitForm: jest.fn(),
    setFormikState: jest.fn(),
    ...options
  };
}

export function getMockedFormikHandles(options: Partial<FormikHandlers> = {}): FormikHandlers {
  return {
    handleSubmit: jest.fn(),
    handleReset: jest.fn(),
    handleBlur: jest.fn(),
    handleChange: jest.fn(),
    ...options
  };
}

export function getMockedField(options: Partial<FormikField> = {}): FormikField {
  return {
    onChange: jest.fn(),
    onBlur: jest.fn(),
    value: '',
    name: '',
    ...options
  };
}
