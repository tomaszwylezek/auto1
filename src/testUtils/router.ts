import { RouteComponentProps } from 'react-router';
import { UnregisterCallback, Href, Location } from 'history';

export function getMockedRouterProps<P = any>(params?: P, locationData?: Partial<Location>) {
  const location: Location = {
    pathname: '',
    hash: '',
    key: '',
    search: '',
    state: '',
    ...locationData
  };

  const props: RouteComponentProps<P> = {
    match: {
      isExact: true,
      params,
      path: '',
      url: ''
    },
    location,
    history: {
      length: 2,
      action: 'POP',
      location,
      push: jest.fn(),
      replace: () => null,
      go: () => null,
      goBack: jest.fn(),
      goForward: () => null,
      block: () => {
        return null as UnregisterCallback;
      },
      createHref: () => {
        const temp: Href = '';
        return temp as Href;
      },
      listen: () => {
        return null as UnregisterCallback;
      }
    },
    staticContext: {}
  };

  return props;
}
