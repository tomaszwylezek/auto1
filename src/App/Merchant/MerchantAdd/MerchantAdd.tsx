import React, { FC } from 'react';
import uuidv4 from 'uuid/v4';

// Models
import { IMerchantAddProps } from './MerchantAdd.types';

import { IMerchantFormValues } from 'App/Merchant/MerchantForm/MerchantForm.types';
import { MerchantFormContainer } from 'App/Merchant/MerchantForm/MerchantForm.container';
import { Button } from 'shared/Button/Button';
import { ButtonTheme } from 'model';

export const MerchantAdd: FC<IMerchantAddProps> = ({ addMerchant, history: { push } }) => {
  return (
    <section>
      <Button theme={ButtonTheme.outlineInfo} to="/">
        Back
      </Button>
      <h2>Add merchant</h2>
      <MerchantFormContainer
        onSubmit={(values: IMerchantFormValues) => {
          Promise.resolve(
            addMerchant({
              id: uuidv4(),
              bids: [],
              ...values
            })
          ).then(() => {
            push('/');
          });
        }}
      />
    </section>
  );
};
