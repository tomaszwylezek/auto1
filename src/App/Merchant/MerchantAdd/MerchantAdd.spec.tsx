import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { MerchantAdd } from './MerchantAdd';

// Models
import { IMerchantAddProps } from './MerchantAdd.types';
import { getMockedRouterProps } from 'testUtils/router';

describe('MerchantAdd component should', (): void => {
  let props: IMerchantAddProps;
  let component: ShallowWrapper<IMerchantAddProps>;

  beforeEach((): void => {
    props = {
      addMerchant: jest.fn(),
      ...getMockedRouterProps()
    };

    component = shallow(<MerchantAdd {...props} />);
  });

  it('render correctly', (): void => {
    expect(component).toMatchSnapshot();
  });
});
