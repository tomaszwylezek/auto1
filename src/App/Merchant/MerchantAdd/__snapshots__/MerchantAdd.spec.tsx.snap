// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MerchantAdd component should render correctly 1`] = `
<section>
  <Button
    theme="outline-info"
    to="/"
  >
    Back
  </Button>
  <h2>
    Add merchant
  </h2>
  <WithFormik(MerchantForm)
    onSubmit={[Function]}
  />
</section>
`;
